woocommerce-gateway-stripe.php

```diff
@@ -732,10 +732,6 @@ public function checkout_update_email_field_priority( $fields ) {
 
 					// Add extra `stripe-gateway-checkout-email-field` class.
 					$fields['billing_email']['class'][] = 'stripe-gateway-checkout-email-field';
-
-					// Append StripeLink modal trigger button for logged in users.
-					$fields['billing_email']['label'] = $fields['billing_email']['label']
-						. ' <button class="stripe-gateway-stripelink-modal-trigger"></button>';
```

Comment:
This will remove the field label and the Stripe Link trigger, both of which we'd like to keep.

We just added back support for Stripe Link in shortcode checkouts in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3380, so we need to keep the trigger button. One viable path is to remove the `button` element from the label field, and add it via JavaScript when we load Stripe Link.

Requested changes:
- keep the email address field label (see screenshot)
- inject the Stripe Link button via JS, outside of the label element, to keep it from appearing in the error message

Screenshot: Email address input label
<img width="183" alt="Screenshot 2024-08-28 at 9 35 23 AM" src="https://github.com/user-attachments/assets/8adcff38-e694-4600-94a2-c1bf614c3454">

Reply:
Thanks for the valuable suggestion @annemirasol.

From which file should I inject the Stripe Link button via JS. I am going through the codebase and still in a confusion about the best approach to append the Stripe Link button.

Reply:
@sapayth Thanks again for taking this on! Here are some things to get started with:

1. You can see how Blocks Checkout injects the Link button here: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/5934d4a269c6de15d4b588b98f03761b85f56bde/client/blocks/upe/hooks.js#L162

2. I would suggest updating Shortcode Checkout's `showLinkButton` to also inject the button, similar to how Block checkout does it, instead of assuming it's already there and just un-hiding it: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/5934d4a269c6de15d4b588b98f03761b85f56bde/client/stripe-link/index.js#L1

3. As Blocks Checkout and Shortcode Checkout will have the same Link button logic now, this fork is no longer needed: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/5934d4a269c6de15d4b588b98f03761b85f56bde/client/stripe-link/index.js#L38

4. After the Link button logic unification, Blocks Checkout will no longer need to have its own `show_button` method, and cleanups can be done.